cmd/humanlog/genny.go

```diff
@@ -90,18 +99,79 @@ func genny(
 
 	now := start
 	for {
-		if err := emitMessage(out, now, src); err != nil {
-			return err
-		}
-		nextArrival := arrivalRateDist.Rand()
-		nextMsgIn := time.Duration(float64(interval) / nextArrival)
 		select {
 		case <-ctx.Done():
 			return nil
-		case <-time.After(nextMsgIn):
+		default:
+			nextArrival := arrivalRateDist.Rand()
+			nextMsgIn := time.Duration(float64(interval) / nextArrival)
+			time.Sleep(nextMsgIn)
 			now = now.Add(nextMsgIn)
+
+			if err := emitLog(out, now, src, format); err != nil {
```

Comment:
this replaces the usage of `emitMessage` but doesn't remove that func, which is now dangling as unused code. can you think of a way to reconcile the two?